File: core/src/main/java/com/andysklyarov/finnotify/data/CurrencyInRubRepository.java
package com.andysklyarov.finnotify.data;

import com.andysklyarov.finnotify.domain.CurrencyInRub;

import java.time.LocalDate;

import io.reactivex.Single;

public class CurrencyInRubRepository {
    private CurrencyInRubDataSource currencyDataSource;

    public CurrencyInRubRepository(CurrencyInRubDataSource dataSource) {
        currencyDataSource = dataSource;
    }

    public Single<CurrencyInRub> getCurrencyOnDate(String currencyCode, LocalDate date) {
        return currencyDataSource.getCurrencyOnDate(currencyCode, date);
    }

    public Single<LocalDate> getLastDate() {
        return currencyDataSource.getLastDate();
    }
}

File: core/src/main/java/com/andysklyarov/finnotify/data/CurrencyInRubRepository.kt
package com.andysklyarov.finnotify.data

import com.andysklyarov.finnotify.domain.CurrencyInRub
import io.reactivex.Single
import java.time.LocalDate

class CurrencyInRubRepository(private val currencyDataSource: CurrencyInRubDataSource) {

    fun getCurrencyOnDate(currencyCode: String, date: LocalDate): Single<CurrencyInRub> {
        return currencyDataSource.getCurrencyOnDate(currencyCode, date)
    }

    fun getCurrencyOnPreviousDate(currencyCode: String, date: LocalDate): Single<CurrencyInRub> {
        return currencyDataSource.getCurrencyOnPreviousDate(currencyCode, date)
    }

    fun getLastDate(): Single<LocalDate>{
        return currencyDataSource.getLastDate()
    }
}